pkg/security/probe/activity_dump.go

```diff
@@ -513,6 +517,11 @@ func (ad *ActivityDump) isEventTypeTraced(event *model.Event) bool {
 	return false
 }
 
+// IsEmpty return true if the dump did not contain any nodes
+func (ad *ActivityDump) IsEmpty() bool {
+	return len(ad.ProcessActivityTree) == 0
```

Comment:
I think this should lock the mutex

Reply:
Suggestion applied on last [commit](https://github.com/DataDog/datadog-agent/pull/15013/commits/a68907931d767167f51519c00567f05042cf1baa), thank you